packages/components/src/dropdown-menu-v2/group-label.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { WordPressComponentProps } from '../context';
+import { DropdownMenuContext } from './context';
+import { Text } from '../text';
+import type { DropdownMenuGroupProps } from './types';
+import * as Styled from './styles';
+
+export const DropdownMenuGroupLabel = forwardRef<
+	HTMLDivElement,
+	WordPressComponentProps< DropdownMenuGroupProps, 'div', false >
+>( function DropdownMenuGroup( props, ref ) {
+	const dropdownMenuContext = useContext( DropdownMenuContext );
+	return (
+		<Styled.DropdownMenuGroupLabel
+			ref={ ref }
+			render={
+				// @ts-expect-error The `children` prop is passed
+				<Text
+					upperCase
```

Comment:
Is there any chance we might want the group label not to be uppercase? Or is this an [intentional design decision](https://wordpress.org/about/philosophy/#:~:text=Decisions%2C%20not%20options)?

Reply:
I assume it's intentional, but i'll let @jameskoster confirm

Reply:
The type should align to a specific style from the outcome of the work in https://github.com/WordPress/gutenberg/issues/64340#issuecomment-2293705925. 

I don't know how that would be built into the component config file though, or consumed by the `Text` component. I plan to open a PR soon to add tokens for the individual properties (font, size, line-height, weight), perhaps we can continue the discussion there?